prowler/providers/aws/services/iam/iam_service.py

```diff
@@ -661,6 +665,27 @@ def __list_tags_for_resource__(self):
                 f"{self.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
 
+    def __get_last_accessed_services__(self):
+        for user in self.users:
+            details = self.client.generate_service_last_accessed_details(Arn=user.arn)
+            response = self.client.get_service_last_accessed_details(
+                JobId=details["JobId"]
+            )
+            while response["JobStatus"] == "IN_PROGRESS":
```

Comment:
Why do we need to do the `while` block?

Reply:
API returns while the job to generate_service_last_accessed_details is ongoing with the JobStatus "IN_PROGRESS", there is a need to call the API until the JobStatus changed from "IN_PROGRESS" to "COMPLETED"